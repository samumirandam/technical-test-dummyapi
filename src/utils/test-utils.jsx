import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import reduxThunk from "redux-thunk";
import { configureStore as configureReduxStore } from "@reduxjs/toolkit";
import { BrowserRouter } from "react-router-dom";

import reducer from "@reducers";

const middlewares = [reduxThunk];
const mockStore = configureStore(middlewares);

const AllTheProviders = ({ children }) => {
  const store = configureReduxStore({
    reducer: reducer,
  });
  const setupStore = { ...store.getState(), ...children.props };
  const compomentStore = mockStore(setupStore);
  return (
    <Provider store={compomentStore}>
      <BrowserRouter>{children}</BrowserRouter>
    </Provider>
  );
};

const customRender = (ui, options) =>
  render(ui, { wrapper: AllTheProviders, ...options });

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
