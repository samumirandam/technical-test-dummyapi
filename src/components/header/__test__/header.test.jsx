import React from "react";
import { render, screen, fireEvent } from "@utils/test-utils";

import Header from "../index";

const defaultProps = {};

const setup = (properties = {}) => {
  const setupStore = { ...defaultProps, ...properties };
  return render(<Header {...setupStore} />, {});
};

describe("Test for Header component", () => {
  test("Should render without errors", () => {
    setup();
    expect(screen.getByText("Iniciar sesiÃ³n")).toBeTruthy();
  });

  test("Should renders the same component", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  test("Should render click in open user profile", () => {
    setup();
    fireEvent.click(screen.getByText("Petgram"));
    expect(global.window.location.pathname).toEqual("/");
  });
});
