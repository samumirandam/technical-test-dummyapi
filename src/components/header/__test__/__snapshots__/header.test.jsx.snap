// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Header component Should renders the same component 1`] = `
<div>
  <header
    class="Header"
  >
    <nav>
      <ul>
        <li>
          <div
            class="Button"
          >
            Iniciar sesi√≥n
          </div>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
